### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'encode' - ERPNext: v13.7.0 (version-13)

### ISSUE THREAD

**Issue by muchai** [frappe/erpnext#26577]
I think it happens when a user has no permission for at least one of the doctypes in the workbench
![image](https://user-images.githubusercontent.com/1822868/126457647-05186be8-5e0e-4882-8f9c-ea18599af636.png)

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 68, in application
    response = frappe.api.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 67, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1172, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 609, in wrapper_fn
    retval = fn(*args, **get_newargs(fn, kwargs))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/desktop.py", line 364, in get_desktop_page
    wspace.build_workspace()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/desktop.py", line 194, in build_workspace
    'items': self.get_charts()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/desktop.py", line 22, in wrapper
    return fn(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/desktop.py", line 299, in get_charts
    if frappe.has_permission('Dashboard Chart', doc=chart.chart_name):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 702, in has_permission
    out = frappe.permissions.has_permission(doctype, ptype, doc=doc, verbose=verbose, user=user, raise_exception=throw)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 26, in inner
    result = func(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 63, in has_permission
    perm = get_doc_permissions(doc, user=user, ptype=ptype).get(ptype)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 118, in get_doc_permissions
    if has_controller_permissions(doc, ptype, user=user) == False :
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 307, in has_controller_permissions
    controller_permission = frappe.call(frappe.get_attr(method), doc=doc, ptype=ptype, user=user)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1172, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/doctype/dashboard_chart/dashboard_chart.py", line 76, in has_permission
    allowed_doctypes = frappe.permissions.get_doctypes_with_read()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 315, in get_doctypes_with_read
    return list(set([p.parent if type(p.parent) == str else p.parent.encode('UTF8') for p in get_valid_perms()]))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 315, in <listcomp>
    return list(set([p.parent if type(p.parent) == str else p.parent.encode('UTF8') for p in get_valid_perms()]))
AttributeError: 'NoneType' object has no attribute 'encode'

**Comment by OneiricIT** [comment]
We have also seen this occur from a v12 upgrade. 

As temp workaround, removed the dashboard from Workspace so pages are accessible.

Also just found it occurs on communication form when selecting To, CC, BB also.

`Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 68, in application
    response = frappe.api.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 67, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1172, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/__init__.py", line 20, in get_contact_list
    match_conditions = build_match_conditions('Contact')
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/reportview.py", line 548, in build_match_conditions
    match_conditions =  DatabaseQuery(doctype, user=user).build_match_conditions(as_condition=as_condition)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/db_query.py", line 607, in build_match_conditions
    doctype_conditions = self.get_permission_query_conditions()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/db_query.py", line 693, in get_permission_query_conditions
    c = frappe.call(frappe.get_attr(method), self.user)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1172, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/contacts/address_and_contact.py", line 86, in get_permission_query_conditions_for_contact
    return get_permission_query_conditions("Contact")
  File "/home/frappe/frappe-bench/apps/frappe/frappe/contacts/address_and_contact.py", line 92, in get_permission_query_conditions
    links = get_permitted_and_not_permitted_links(doctype)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/contacts/address_and_contact.py", line 122, in get_permitted_and_not_permitted_links
    allowed_doctypes = frappe.permissions.get_doctypes_with_read()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 315, in get_doctypes_with_read
    return list(set([p.parent if type(p.parent) == str else p.parent.encode('UTF8') for p in get_valid_perms()]))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/permissions.py", line 315, in <listcomp>
    return list(set([p.parent if type(p.parent) == str else p.parent.encode('UTF8') for p in get_valid_perms()]))
AttributeError: 'NoneType' object has no attribute 'encode'`
 
![image](https://user-images.githubusercontent.com/22227933/126916361-01e01f43-a1e4-4767-a023-8323e016189f.png)

**Comment by OneiricIT** [comment]
For the contacts, have found it is system wide even for Administrator account with all roles. Even reset permissions back to default and getting the same.

![image](https://user-images.githubusercontent.com/22227933/126917077-4d507a36-b725-4cf3-a208-7468fdc3a626.png)

**Comment by OneiricIT** [comment]
@muchai I resolved my issue - I went through Role Permission Manager resetting doctypes I had set permissions on back to original. I referenced table tabCustom DocPerm for the doctypes affected (didn't remove entries via DB, however should have to make faster) - after cleaning out entries listed against specific doctypes, I found there were many entries still in the table which were not against any doctypes - clearing these fixed it for me. I also spun up a new instance on v13 with a clean site, checked DB and found entries were created so exported and imported this over with still no issue.

Note: running bench command to restore permissions on site did not affect the above, as did this originally and still had the issue.
